Framework/Algorithms/src/PDFFourierTransform2.cpp

```diff
@@ -197,13 +196,13 @@ size_t PDFFourierTransform2::determineMaxIndex(double max, const std::vector<dou
     max = X.back();
   }
 
-  // get pointers for the data range
-  auto iter = std::lower_bound(X.begin(), X.end(), max);
+  // get index for the max from the X-range, tightening the range if a partial x bin value is provided
+  auto iter = std::upper_bound(X.begin(), X.end(), max) - 1;
   size_t max_index = std::distance(X.begin(), iter);
 
   // go to first non-nan value
   auto back_iter = std::find_if(Y.rbegin(), Y.rend(), static_cast<bool (*)(double)>(std::isnormal));
-  size_t first_normal_index = Y.size() - std::distance(Y.rbegin(), back_iter) - 1;
+  size_t first_normal_index = Y.size() - std::distance(Y.rbegin(), back_iter);
```

Comment:
This works fine now for bin edge data. It is also fine for point data although the reason why is a bit subtle. If there's a point dataset like this: 
```
x=[1,2,3], y=[100,101,nan]
```
...the logic sets first_normal_index=3-1=2 ie it leaves max_index pointing at the final point. This is fine because the integration on point data sets is left leaning ie it looks at the Y value on the left of each interval so the final y value never gets used

Please can you add a comment about this? Doesn't need to be as verbose as I've just been (!) eg the logic applies to both bin edge and point data because the integration for points uses the Y value on the left of each interval.

Reply:
Does this mean we can support point data now and we can remove the warning here? Are we happy to just leave this weird commented out density section for now???https://github.com/mantidproject/mantid/blob/cb21841641eb26bc1b0966d6417354dbbfaa6f7d/Framework/Algorithms/src/PDFFourierTransform2.cpp#L381-L397

Reply:
> Does this mean we can support point data now and we can remove the warning here? Are we happy to just leave this weird commented out density section for now???
> 
> https://github.com/mantidproject/mantid/blob/cb21841641eb26bc1b0966d6417354dbbfaa6f7d/Framework/Algorithms/src/PDFFourierTransform2.cpp#L381-L397

I think we should leave the warning and the commented out code alone. I think the specific parts we've looked at work OK for point data but I'm not 100% sure about the rest of the unit

Reply:
I've just put a comment about this in the `determineMaxIndex()`, should there also be an equivalent in `determineMinIndex()`?